### PR TITLE
VMware: Add 'connected' option for virtual network adapters.

### PR DESCRIPTION
**Pull request by native0090**
##### SUMMARY

Added the option to connect or disconnect virtual network adapters in vmware_guest when reconfiguring 
networks. Before it was not possible to connect/disconnect a virtual network adapter.

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

vmware_guest

##### ANSIBLE VERSION

```paste below
ansible 2.8.0.dev0 (devel 812f439d11)
```

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: lib/ansible/modules/cloud/vmware/vmware_guest.py [hunk 1]
```diff
@@ -273,6 +273,7 @@
     - ' - C(dvswitch_name) (string): Name of the distributed vSwitch.
           This value is required if multiple distributed portgroups exists with the same name. version_added 2.7'
     - ' - C(start_connected) (bool): Indicates that virtual network adapter starts with associated virtual machine powers on. version_added: 2.5'
+    - ' - C(start_connected) (bool): Connects or disconnects the virtual network adapter.'
```
Comment (Akasurde) [inline lib/ansible/modules/cloud/vmware/vmware_guest.py]: Could you please rephrase this as - 

```
- ' - C(connected) (bool): Connects or disconnects the virtual network adapter. version_added: 2.8'
```

**Review by Akasurde — CHANGES_REQUESTED** [review]
Thanks for the contribution. Could you please add test case for this change ?

**Comment by native0090** [comment]
How would I add a test case for this? There don't seem to be any unit tests for vmware.

**Comment by Akasurde** [comment]
@mpk93 Integration tests for VMware can be found [here](https://github.com/ansible/ansible/tree/devel/test/integration/targets/vmware_guest/tasks)

**Comment by native0090** [comment]
@Akasurde Thanks. Could you please review my integration test, as I am not very familiar with it.

File: test/integration/targets/vmware_guest/tasks/network_connected.yml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+- debug: var=vcsim_instance
+
+- debug: var=vmlist
+
+- set_fact:
+   vm1: "{{ vmlist['json'][0] }}"
+
+- debug: var=vm1
+
+- name: disconnect network
```
Comment (Akasurde) [inline test/integration/targets/vmware_guest/tasks/network_connected.yml]: Perform same operation again and check status of this task, this will test the idempotency of this feature.

File: test/integration/targets/vmware_guest/tasks/network_connected.yml [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+- debug: var=disconnect_network
+
+- name: assert that no errors occured
+  assert:
+    that:
+        - "not disconnect_network.failed"
+
+- name: connect network
+  vmware_guest:
```
Comment (Akasurde) [inline test/integration/targets/vmware_guest/tasks/network_connected.yml]: Same here.

**Comment by native0090** [comment]
Not sure if you meant that by testing idempotency. As I am not sure if I can assume that the network is connected when I first test the VM.

**Comment by native0090** [comment]
@Akasurde does this PR need more work?

File: lib/ansible/modules/cloud/vmware/vmware_guest.py [hunk 4]
```diff
@@ -272,6 +272,7 @@
     - ' - C(mac) (string): Customize MAC address.'
     - ' - C(dvswitch_name) (string): Name of the distributed vSwitch.
           This value is required if multiple distributed portgroups exists with the same name. version_added 2.7'
+    - ' - C(connected) (bool): Connects or disconnects the virtual network adapter. version_added: 2.8'
```
Comment (Akasurde) [inline lib/ansible/modules/cloud/vmware/vmware_guest.py]:
```suggestion
    - ' - C(connected) (bool): Connects or disconnects the virtual network adapter. version_added: 2.9'
```

File: test/integration/targets/vmware_guest/tasks/network_connected.yml [hunk 5]
```diff
@@ -0,0 +1,135 @@
+# Test code for the vmware_guest module.
+# Copyright: (c) 2018, Maximilian Prahl-Kamps <maximilian.prahl-kamps@fkie.fraunhofer.de>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
```
Comment (Akasurde) [inline test/integration/targets/vmware_guest/tasks/network_connected.yml]: Add this test here as well https://github.com/ansible/ansible/blob/1462fd740b279ebe7dd791c8cac338a36ad413f6/test/integration/targets/vmware_guest/tasks/main.yml#L47

**Review by Akasurde — CHANGES_REQUESTED** [review]
Could you please review new comments ? Thanks.

**Comment by goneri** [comment]
hi @mpk93, your PR includes =~ 100 unrelated commits. Could you please base it or prepare a new PR.
